src/main/java/it/eng/idsa/dataapp/service/impl/SelfDescriptionServiceImpl.java

```diff
@@ -40,27 +39,21 @@ public class SelfDescriptionServiceImpl implements SelfDescriptionService {
 	}
 	private RestTemplate restTemplate;
 	private ECCProperties eccProperties;
-	private ThreadService threadService;
 
-	public SelfDescriptionServiceImpl(RestTemplateBuilder restTemplateBuilder, ECCProperties eccProperties,
-			ThreadService threadService) {
+	public SelfDescriptionServiceImpl(RestTemplateBuilder restTemplateBuilder, ECCProperties eccProperties) {
 		this.restTemplate = restTemplateBuilder.build();
 		this.eccProperties = eccProperties;
-		this.threadService = threadService;
 
 	}
 
 	@Override
 	public Connector getSelfDescription(Message message) {
 		URI eccURI = null;
 
-		int eccPort = (Boolean.TRUE.equals(((Boolean) threadService.getThreadLocalValue("wss"))))
-				? eccProperties.getWssSelfDescriptionPort()
-				: eccProperties.getPort();
 		try {
-			eccURI = new URI(eccProperties.getProtocol(), null, eccProperties.getHost(), eccPort,
-					eccProperties.getSelfdescriptionContext(), null, null);
-			logger.info("Fetching self description from ECC {}.", eccURI.toString());
+			eccURI = new URI(eccProperties.getProtocol(), null, eccProperties.getHost(), eccProperties.getSelfdescriptionPort(),
+					"/internal/sd", null, null);
```

Comment:
Why not extracted to constant at the declaration part? :)

Reply:
I'll do it right away :)